metis-remove-data/src/main/java/eu/europeana/metis/remove/discover/AbstractOrphanIdentification.java

```diff
@@ -37,19 +38,29 @@
   enum DiscoveryMode {
 
     /**
-     * This mode removes all orphans and their descendants.
+     * This mode discovers only identified orphans or their descendants that have no children. The
+     * deleted status of the orphan or its children is not taken into account. Note: in case that
+     * we're removing orphans, we should only use this mode if we intend to completely remove
+     * orphans from the history (rather than mark them as deleted).
      */
-    DISCOVER_ALL_ORPHANS(node -> true),
+    ORPHANS_WITHOUT_DESCENDANTS(node -> node.getChildren().isEmpty(), false),
```

Comment:
Is orphan the right terminology here? Aren't those leafs??

Reply:
I wanted to make a distinction between leafs (in the context of the tree) or orphans (which exists outside the context of trees). The main methods don't know about the tree and in their context the word 'leaf' may not make sense.

Reply:
Let me know if you can find a better naming for that here, otherwise if not we can keep it as is( since I managed to make sense out of it eventually :D )